hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/AWSCredentialProviderList.java

```diff
@@ -330,4 +332,27 @@ public void close() {
   public int size() {
     return providers.size();
   }
+
+
+  /**
+   * Translate an exception if it or its inner exception is an
+   * {@link CredentialInitializationException}.
+   * If this condition is not met, null is returned.
+   * @param path path of operation.
+   * @param throwable exception
+   * @return a translated exception or null.
+   */
+  public static IOException maybeTranslateCredentialException(String path,
```

Comment:
why do we need to do this? I get the AuditFailures are getting wrapped in a SDKClient exception, what's wrapping CredentialInitializationException?

Reply:
the audit and CredentialInitializationExceptions are both SdkClientException. they may both be wrapped by async handler, and while I didn't try to replicate it, if it happened in the same request I'd expect the same failure. [and not expected in sync s3 client].